Experiment to test diagnostic noncompliance propagation

korourke5 opened the pull request:
in README. Created scenario files from prepare.py. Edited run_iteration.sh and summarize.py for new
AWS S3.

# CRCsim PR Template

## Description
<!-- Provide a brief description of the changes in this PR -->

## PR Type
<!-- Mark the appropriate option with an [x] (no spaces around the x) -->
- [ ] Feature (intended to merge)
- [X] Experiment (not intended to merge)

<!-- If this is an experiment PR, delete the following section and complete the experiment-specific sections of this template. -->
<!-- If this is a feature PR, complete the following section and delete the experiment-specific sections. -->

## Experiment review checklist
<!-- Make sure all items are checked before submitting the PR -->
- [X] This experiment branch has the `exp-` prefix
- [X] This experiment uses the appropriate `crcsim` commit hash (typically, the latest commit in the `main` branch). This ensures that the experiment uses the latest version of the model.
- [X] This experiment uses the correct baseline `parameters.json` file
    - [X] Correct screening start and end ages for the experiment
    - [X] Latest values of all calibrated parameters
    - [X] All necessary tests and latest test parameters
    - [X] Latest cost parameters
- [ ] My `prepare.py` script applies incidence rate ratio (IRR) adjustment if appropriate for this experiment - NOT applied
- [X] I have run `prepare.py` and:
    - [X] Confirmed that all scenarios necessary for this experiment were created
    - [X] Confirmed that no extraneous scenarios are in the `scenarios/` directory (eg, uncommitted local changes from a previous experiment)
    - [X] Spot-checked a sample of scenario parameter files to ensure that my scenario creation logic works as expected
- [X] I have updated `crcsim/experiment/README.md` with detailed information about the experiment's goals, scenarios, and corresponding AWS objects
- [X] I am opening this PR as a **Draft** PR

## Experiment review process
<!-- No checkboxes here, because this happens after you submit the PR -->
*All experiments must follow each step of this review process.*
1. Contributor checks all items in experiment review checklist
1. Contributor opens **Draft PR** and tags a collaborator to review
1. Reviewer conducts a thorough review, including pulling experiment branch, running `prepare.py`, and spot-checking scenarios
1. Contributor and reviewer address any issues identified during review
1. Reviewer explicitly approves PR
1. Contributor builds image and pushes to ECR, copies experiment files to S3, and runs experiment
1. Contributor analyzes experiment (eg `crcsim/experiment/summarize.py`)
1. Contributor pushes results summary, so we have documentation of detailed results (eg `crcsim/experiment/summary/summarized.xlsx`)
1. Contributor updates experiment README with a detailed summary of results
1. Contributor closes this PR with an informative comment (eg, brief summary of results)

**Note that experiment PRs are never merged into `main`!** The PR is closed, and the experiment is maintained as a separate branch. That's why we keep all experiment PRs as drafts.

korourke5 committed:
Altered Prepare.py to run experiment described
in README. Created scenario files from prepare.py.
Edited run_iteration.sh and summarize.py for new
AWS S3.

sandypreiss commented on crcsim/experiment/prepare.py:
```diff
@@ -130,90 +137,102 @@ def create_scenarios() -> List:
+
+    propagate_diagnostic_noncompliance = {
+        "Diagnostic_Noncompliance": True,
+        "NO_Diagnostic_Noncompliance": False,
```
It might make the summary sheets easier to read if you named these something like "propagation_on" and "propagation_off".

sandypreiss commented on crcsim/experiment/prepare.py:
```diff
@@ -130,90 +137,102 @@ def create_scenarios() -> List:
-                transform_treatment_cost(
-                    "3", "initial", low_initial_stage_3_treatment_cost
+            for (
+                diagnostic_noncompliance,
+                diagnostic_noncompliance_bool,
```
Naming is also a bit confusing here - perhaps `propagation_name` and `propagation_value`?

(It's usually not a good idea to include a variable's type in its name.)

sandypreiss reviewed (changes requested):
I think I see the issue here. This experiment is still installing `crcsim` from an older commit. (It looks like you copied the commit hash from `exp-fqhc-diagnositic-compliance-comparison`.)

Experiments use whatever version of `crcsim` is installed in the Docker image. That version is defined in `pyproject.toml`, which creates `uv.lock`, which is used to install the project in the `Dockerfile`.

So, even though this experiment branch has the new feature in the local code, it's running the experiment with an older version of `crcsim`. That means the `propagate_diagnostic_noncompliance` parameter is never used.

To resolve this: 
1. Update the `crcsim` commit hash in `pyproject.toml` to the latest commit from `main` (now that you've merged the noncompliance propagation branch in). You can get that by opening the commit history from the repo landing page. Just to be sure, it's `c4e08bffba34c07d18b784c0632de506a36611e8`.
2. Update the comment in `pyproject.toml` to explain which commit you're installing from and why.
3. From `crcsim/experiment`, run `uv lock` to update the lockfile.
4. Proceed with the normal steps to run the experiment (including rebuild and push).

I also added a few small style suggestions.

korourke5 committed:
update with changes from PR and to S3 bucket name

korourke5 commented:
@sandypreiss Those updates should be applied.

korourke5 committed:
Added hispanic male and female death rates

korourke5 commented:
@sandypreiss I have updated the scenarios here with the hispanic male and female parameters added and removed extra low and low cost scenarios since I don't believe that will impact this analysis.

korourke5 committed:
Update prepare.py to include control scenarios

korourke5 committed:
Updated prepare.py to include new scenarios

sandypreiss commented:
This is looking good. Have you run it yet? Are we seeing the effect we'd expect from diagnostic noncompliance propagation now?

korourke5 commented:
I am currently running it. I had a bit of a problem running it last week because of the race specific mean_duration_clin but have rectified it. I hope to have results to send over tomorrow.

sandypreiss commented:
This was run successfully. Results need to be pushed and a brief summary added to experiment readme.

korourke5 committed:
Added new scenario files

korourke5 committed:
Updated files to most recent version

korourke5 committed:
Updated results in README

korourke5 commented:
Updated with results and scenario files.
